### PR TITLE
 Remove remainings of actorFor, #26190

### PR DESCRIPTION
**Pull request by patriknw**
## Purpose

Remove deprecated `actorFor`

## References

References #26190

## Changes

* remove `actorFor` methods
* change/remove tests that still used it
* DeathWatch had some special logic for ActorRef with `undefinedUid` (0) that
  could be removed now

## Background Context

Deprecated since 2.2 so high time to remove

### CODE REVIEW

File: akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala [hunk 1]
```diff
@@ -38,8 +39,10 @@ private[akka] trait DeathWatch { this: ActorCell =>
 
   override final def watchWith(subject: ActorRef, msg: Any): ActorRef = subject match {
     case a: InternalActorRef =>
+      if (subject.path.uid == 0)
+        println(s"# undefinedUid watchWith $self -> $subject ${subject.getClass.getName}") // FIXME
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala]: I'll remove these printlns but want to verify if they are triggered by any of the tests first

File: akka-remote/src/test/scala/akka/remote/RemoteDeathWatchSpec.scala [hunk 2]
```diff
@@ -79,9 +79,11 @@ akka {
 
   "receive Terminated when watched node is unknown host" in {
     val path = RootActorPath(Address(protocol, system.name, "unknownhost", 2552)) / "user" / "subject"
+
+    @silent
+    val watchee = RARP(system).provider.resolveActorRef(path)
+
```
Comment (johanandren) [inline akka-remote/src/test/scala/akka/remote/RemoteDeathWatchSpec.scala]: Moved just to be able to silence?

**Review by johanandren — APPROVED** [review]
LGTM, _finally!_

File: akka-actor/src/main/mima-filters/2.5.x.backwards.excludes [hunk 3]
```diff
@@ -4,3 +4,24 @@ ProblemFilters.exclude[MissingClassProblem]("akka.actor.ActorDSL$")
 ProblemFilters.exclude[MissingClassProblem]("akka.actor.ActorDSL")
 ProblemFilters.exclude[MissingClassProblem]("akka.actor.ActorDSL$*")
 ProblemFilters.exclude[MissingClassProblem]("akka.actor.dsl.*")
+
+# #26190 remove actorFor
+ProblemFilters.exclude[DirectMissingMethodProblem]("akka.actor.ActorCell.actorFor")
```
Comment (chbatey) [inline akka-actor/src/main/mima-filters/2.5.x.backwards.excludes]: why is each one duplicated?

File: akka-remote/src/main/mima-filters/2.5.x.backwards.excludes [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# #26190 remove actorFor
+ProblemFilters.exclude[DirectMissingMethodProblem]("akka.remote.RemoteActorRefProvider.actorFor")
```
Comment (chbatey) [inline akka-remote/src/main/mima-filters/2.5.x.backwards.excludes]: here too

**Review by chbatey — APPROVED** [review]

File: akka-actor/src/main/mima-filters/2.5.x.backwards.excludes (hunk 3 above)
Comment (patriknw) [reply akka-actor/src/main/mima-filters/2.5.x.backwards.excludes]: don't know, maybe it was several warnings and I just copied - I'll remove the duplicates

File: akka-remote/src/test/scala/akka/remote/RemoteDeathWatchSpec.scala (hunk 2 above)
Comment (patriknw) [reply akka-remote/src/test/scala/akka/remote/RemoteDeathWatchSpec.scala]: no good reason

**Comment by patriknw** [comment]
rebased and fixed review comments, will merge after validation

**Pull request merged**